src/main/java/Shop.java

```diff
@@ -0,0 +1,23 @@
+public class Shop {
+    public static void main(String[] args) {
+        Category category1 = new Category("Pikantne", "Bardzo ostre");
+        Category category2 = new Category("Łagodne", "Delikatny smak");
+
+        Product product1 = new Product("Musztarda", 10, "Bardzo ostra, francuska", category1);
+        SpecialOffer offer1 = new SpecialOffer(product1, "Promo w tym tygodniu", 7, product1.price);
```

Comment:
jako zniżkę nie powinieneś przekazywać `product1.price`, tylko np. `0.2`, dzięki czemu będzie to można rozumieć tak: "oferta specjalna na musztardę, 20% taniej"